From 1ec5c98d80de97f9e962c5627e1a0e6096099894 Mon Sep 17 00:00:00 2001
From: Daniel Scharrer <daniel@constexpr.org>
Date: Wed, 28 Jul 2021 19:56:31 +0200
Subject: [PATCH] Fix #include inside boost namespace

The existing code fails to build if <utility> was not already included.
---
 include/boost/math/tools/mp.hpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/boost/math/tools/mp.hpp b/include/boost/math/tools/mp.hpp
index 35565646f..dc8440988 100644
--- a/include/boost/math/tools/mp.hpp
+++ b/include/boost/math/tools/mp.hpp
@@ -13,6 +13,7 @@
 
 #include <type_traits>
 #include <cstddef>
+#include <utility>
 
 namespace boost { namespace math { namespace tools { namespace meta_programming {
 
@@ -338,7 +339,6 @@ using mp_remove_if_q = mp_remove_if<L, Q::template fn>;
 // Index sequence
 // Use C++14 index sequence if available
 #if defined(__cpp_lib_integer_sequence) && (__cpp_lib_integer_sequence >= 201304)
-#include <utility>
 template<std::size_t... I>
 using index_sequence = std::index_sequence<I...>;
 
-- 
2.31.1

